source/base/function_parser.cc

```diff
@@ -158,27 +158,25 @@ FunctionParser<dim>::init_muparser() const
   // check that we have not already initialized the parser on the
   // current thread, i.e., that the current function is only called
   // once per thread
-  Assert(fp.get().size() == 0, ExcInternalError());
+  ParserData &data = parser_data.get();
```

Comment:
So here is the functional change: Have only one call to `get()` per function call. To make that work, if I understand right, you need to avoid using multiple thread local variables and so you put it all into one structure?

Reply:
Yup!